### DISCUSSION TITLE
Moving to the Open Wallet Foundation

### DISCUSSION THREAD

**Discussion by karimStekelenburg** [openwallet-foundation/credo-ts#1586] [Announcements]
Hi Community!

The recent discussions in the Aries Working Group and the .NET framework's move to the Open Wallet Foundation (OWF) have prompted us, the active maintainers of Aries Framework JavaScript (AFJ), to consider the framework's future, and drove us to propose to to move the framework to OWF. Here's why.

AFJ's mission has always been to make SSI (Self-Sovereign Identity) accessible to anyone interested in adopting the technology. As decentralized identity technologies are being adopted by different governments and organizations, various standards are being used. We believe that AFJ is most valuable as a generic SSI framework that supports a wide range of standards. Therefore, we have expanded AFJ's capabilities by adding support for other non-Aries standards, such as OpenID4VC and JSON-LD, over the past year.

However, we think that AFJ has a "branding problem." Due to its name and association with other Hyperledger projects, such as Indy, AnonCreds, and Fabric, most people assume that the framework is limited to the Hyperledger ecosystem. This makes it difficult for the framework to be adopted in jurisdictions that rely on non-Aries standards and also limit its appeal to new contributors. We believe that moving the framework to OWF will better communicate its purpose, has the potential to attract a wider audience, increase contributors, and overall adoption of AFJ.

We understand that AFJ has many stakeholders who require interoperability with AFJ based on the Aries Interop Profiles (such as ACA-Py). Therefore, we want to emphasize that we will continue to work closely with the Aries community and support the standards on which Aries is based (AnonCreds, DIDComm, etc.).

This is only a proposal. As maintainers, we value the views and opinions of the community and will take any objections seriously. If no formal objections are raised within the next 14 days, we will initiate the process of moving AFJ to OWF.

To make a formal objection, please reply to this post, clearly stating that you object and why.

Kind regards,

Ariel, Jakub, Berend, Timo, Karim

**Comment by ankurdotb** [comment]
As one of the supported methods in AFJ, I'm broadly supportive of this if there's community consensus. There's a few comments/thoughts I have:

1. Would this become the de-facto SDK under OWF for TypeScript/JavaScript users? i.e., is the idea to make this a one-stop-shop SDK? While I know AFJ has been adding a few things like OIDC4VC, JSON-LD/JWT, there's other comms protocols, VC formats etc. One route to go down is to use AFJ as the base for a "super SDK" that handles all use cases (possibly with modules). The other is to stick to Aries / AnonCreds as the base, with some of the additional formats/protocols supported on top. Or to summarise, as a developer, is the idea that I can *just* use `@owf/<new-AFJ-name` for everything, or will I need to use `@owf/<new-AFJ-name` + `@veramo/**` for anything AFJ doesn't cover?
2. Would this be a fork of AFJ or actually shifting the project to a different Github organisation? If there's broad consensus, I'd recommend the latter since it retains history in subtle/different ways. I understand though that people might want to keep both repos around.
3. Will the existing NPM packages be marked as deprecated with a redirect to the new packages? I guess that depends on above /  whether there are maintainers for both. In case the whole community does shift, my preference would be to leave the current packages published, but mark them as deprecated, which still allows them to be used in current libraries/projects, but explicitly letting people know where the development activity has shifted.

Also, to be completely clear, I'm agnostic to whether there's a fork and there are two projects, AFJ remains as-is under Hyperledger, or a full shift to OWF. My intention with the above to understand as a developer whether I need to deal with one project or two.

**Comment by samuelmr** [comment]
Wow!
I personally support the proposal. If other Aries frameworks followed this path, we could have ...
- ... one framework with interoperable implementations/libraries for many programming languages ...
-  ... enabling privacy-preserving use cases with AnonCreds + optional ledger, _and_ DIDComm connections, _and_ eIDAS-compliant technologies ...
- ... maintained and supported by an organization dedicated to digital wallets.

**Comment by TimoGlastra** [comment]
Hey @ankurdotb. Good questions, i'll try to answer them one by one:

> _1. Would this become the de-facto SDK under OWF for TypeScript/JavaScript users? i.e., is the idea to make this a one-stop-shop SDK? While I know AFJ has been adding a few things like OIDC4VC, JSON-LD/JWT, there's other comms protocols, VC formats etc. One route to go down is to use AFJ as the base for a "super SDK" that handles all use cases (possibly with modules). The other is to stick to Aries / AnonCreds as the base, with some of the additional formats/protocols supported on top. Or to summarise, as a developer, is the idea that I can just use @owf/<new-AFJ-name for everything, or will I need to use @owf/<new-AFJ-name + @veramo/** for anything AFJ doesn't cover?_

I don't think we're in the position where we can claim this will be the de-facto SDK under OWF for JS/TS. We've had positive feedback on AFJ moving to OWF from several people involved with OWF, but we still have to make a formal proposal and get it accepted by TAC.

We've been thinking a lot more lately about the purpose and usage of AFJ, and what we want the framework to be. I think our current thought is that we'd like it to be a very easy and accessible framework that connects all the different standards involved in SSI/DI. Previously the approach has been to implement all these standards ourselves within the framework, but with the expanded availabaility of libraries we'd like to move more towards it connecting different libraries. 

This could mean for example that:
- OID4VCI and PEX support is provided by Sphereon's libraries
- JWT VCs support is provided by DIFs implementation
- JSON-LD support is provided by Digital credentials (https://github.com/digitalcredentials/linked-data-integrity/)
- did resolver implementation is provided by DIFs library
- SD-JWT is used from another library under OWF (e.g. https://github.com/berendsliedrecht/sd-jwt-ts).
- DIDComm support with implementation of certain protocols will be extracted into separatie libraires/repos.

The framework would then add the statefullness and 'glue' everything together. The other libraries can then be re-used more, and you don't have to use the framework if you just want to resolve a did for example.

The goal is definitely not to stick to AnonCreds / Aries as the base. It will be one of the protocols and credential formats supported. This is also a reason for proposing the move to OWF -- to make it less bound to any specific credential format or protocol.

> Or to summarise, as a developer, is the idea that I can just use @owf/<new-AFJ-name for everything

Yes I think that would be our goal. Maybe there's ways to combine the efforts put into veramo and AFJ (sharing underlying libraries could help), that is something to explore. Will there even be a need to have two frameworks?


> _2. Would this be a fork of AFJ or actually shifting the project to a different Github organisation? If there's broad consensus, I'd recommend the latter since it retains history in subtle/different ways. I understand though that people might want to keep both repos around._

This depends a bit on what the community and the foundations would like. Ideally I'd say we move the repository over. We intend to keep interop with other Aries frameworks and keep supporting all the features that AFJ does today -- so I don't think there'd be a need to fork / keep two repositories alive.

> _3. Will the existing NPM packages be marked as deprecated with a redirect to the new packages? I guess that depends on above / whether there are maintainers for both. In case the whole community does shift, my preference would be to leave the current packages published, but mark them as deprecated, which still allows them to be used in current libraries/projects, but explicitly letting people know where the development activity has shifted._

Yes, ideally we keep the old packages, but mark them as deprecated and point to the new place.

**Comment by swcurran** [comment]
Regards the following — please do not fork AFJ. I think that would be a mess.  If it is going to OWF, it should be moved, not forked. It is under very active development and use, and a fork would create significant problems IMHO.  This is a very different situation than the .NET code that had been unmaintained in Hyperledger for a long time.  We really need the move to go smoothly, with as little disruption to the ongoing work as possible.

> 2. Would this be a fork of AFJ or actually shifting the project to a different Github organisation? If there's broad consensus, I'd recommend the latter since it retains history in subtle/different ways. I understand though that people might want to keep both repos around.

> This depends a bit on what the community and the foundations would like. Ideally I'd say we move the repository over. We intend to keep interop with other Aries frameworks and keep supporting all the features that AFJ does today -- so I don't think there'd be a need to fork / keep two repositories alive.

**Comment by ankurdotb** [comment]
How would we reach a conclusion on this? Will this be put to a vote by the community?

**Comment by TimoGlastra** [comment]
Our current thinking was that we will post this, and have a two week (which has now elapsed) window for format objections. We haven't received any format objections on this post, and thus we'd like to formally start the process of submitting a proposal to the Open Wallet Foundation to move Aries Framework JavaScript (under a new name) to the Open Wallet Foundation.

Instead of putting it to a vote, we've chosen to go with a two week formal objection period. So it could be seen as no formal objection = accept proposal to move to OWF, formal objection = do not accept proposal to move to OWF).

We will spend more time on this during the AFJ Working Group call this Thursday

**Reply by ankurdotb** [reply]
Given the scale of the move, I wonder if 2 weeks is enough. That timescale is short enough that key stakeholders might be on holiday. For sure, discussing on the Aries Framework JavaScript Working Group is a good idea. I would also try and socialise this idea on perhaps the DIF/W3C/OWF mailing lists since I'm not sure everyone watches the issues here, and key stakeholders might not have noticed. (I discovered it, for example, because I follow you on X/Twitter, and came across it there. If I hadn't looked at Twitter that day, I could have _easily_ missed it.)

Socialising the link to the discussion (and encouraging them to respond here, rather than directly on the mailing list, could help ensure this doesn't come as a surprise to anyone. The SSI industry is small enough that you'll hit 95% of the people who care via those mailing lists.

It will just help prevent any surprise/bad blood later on. :) FWIW, like I said before, I'm supportive of the idea.

**Reply by TimoGlastra** [reply]
That's a good suggestion. @karimStekelenburg has posted it on the Aries mailing lists, but we didn't post it to the other mailing lists. Mainly because AFJ is not used that much in the W3C / DIF circles. 

I'll post it to the OWF mailing list (now that we also haven't received any formal objections from the Aries community), and give it some more time

**Comment by ajile-in** [comment]
Hi @TimoGlastra , @karimStekelenburg - The argument & proposal make sense, I support the *move* of AFJ to OWF.

**Comment by TimoGlastra** [comment]
The proposal has now been submitted to the Open Wallet Foundation Project Proposals Repository: https://github.com/openwallet-foundation/project-proposals/pull/26

**Comment by TimoGlastra** [comment]
Hello everyone, the proposal to move Aries Framework JavaScript to the Open Wallet Foundation has been unanimously accepted, and we're now the first growth project under the Open Wallet Foundation :tada: 

The AFJ and EXT repositories have already been moved to the Open Wallet Foundation Github:
- https://github.com/openwallet-foundation/agent-framework-javascript
- https://github.com/openwallet-foundation/agent-framework-javascript-ext

The documentation site is in the process of being moved, but for now the documentation is still available at https://github.com/hyperledger/aries-javascript-docs and https://aries.js.org.

You should be automatically redirected when trying to access the Hyperledger Github URL, but it's good to update your references and links.

 With the move to the Open Wallet Foundation, we're also moving the **discussion for Aries Framework JavaScript to the Open Wallet Foundation discord.**

Please join https://discord.gg/openwalletfoundation, and join the agent-framework-js channel: https://discord.com/channels/1022962884864643214/1179453305856991263

The name for the project is still under discussion, and it's currently hosted as Agent Framework JavaScript. Please track the discussion created by @genaris for the status of this: https://github.com/openwallet-foundation/agent-framework-javascript/discussions/1668

We're in the process of moving the meetings, notes, and updating the publishing, CI, READMEs etc.. to reflect the new home and name. This will take a while to complete, and so please know it can be a bit chaotic in the coming weeks while we perform the needed work to facilitate the change. Once the migration has been completed, we'll do another formal announcement, but we want to make sure everyone is aware of what's happening. **We'll make sure to update this thread**

Nothing will change for now, and we'll continue the work as we've been doing for the last couple of years

**Comment by TimoGlastra** [comment]
Done :)
